### DISCUSSION TITLE
Re-connection issue while hosting on Oracle Cloud

### DISCUSSION THREAD

**Discussion by kruleucc** [jagrosh/MusicBot#823] [Troubleshooting]
Wanted to open a discussion rather than issue for this little problem @MichailiK 

Here is the debug command output you wanted me to upload.
[debug_information.txt](https://github.com/jagrosh/MusicBot/files/7177450/debug_information.txt)

**Comment by MichailiK** [comment]
Do you also have any console logs?

**Reply by kruleucc** [reply]
In what logs would you be interested? Haven't done any yet.

**Reply by MichailiK** [reply]
The... logs from the standard output... these ones:

![image](https://user-images.githubusercontent.com/39029839/133609484-ac47ce65-a38a-4b0f-b40d-dc42c6bb7dde.png)

**Reply by kruleucc** [reply]
Oh those, they are all good I think
![image](https://user-images.githubusercontent.com/3966846/133609806-9710a48c-b111-42e0-b53c-963345ecf4c1.png)

I will check the logs next morning as the disconnection happens at night. Also I found out that the process stays running on the vm but bot goes offline on Discord. After running the jar again I found out that I have 2 jars running.

**Comment by YJSoft** [comment]
If you're using Micro E1 instance, Micro E1 Instances are limited to 50Mbps(about 6.25MB/s)
So it would be not enough for running from three discord servers.

If not, please check if discord voice server and server region matchs.

**Reply by MichailiK** [reply]
YouTube's audio streams seem to be generally around 140 KB/s based on [this music video](https://www.youtube.com/watch?v=MWASeaYuHZo)'s opus stream. That's 0.14 MB/s. JMusicBot doesn't download the entire video either, it only buffers the next few seconds. Mathematically, JMusicBot would theoretically be able able to handle up to `6.25 / 0.14 = 44` audio streams. The bottleneck more likely lies on the processor not being handle to deal with multiple streams simultaneously.

Finally, this is about the bot losing connection to Discord & not re-connecting to Discord during the night.

**Reply by YJSoft** [reply]
I almost forgot...
but Now I remember why I stopped using JMusicBot with Oracle. Bot stopped without clear reason, so I moved to lightsail and It's no longer stopped.

**Comment by kruleucc** [comment]
Here is the nohup.out file from the cloud: 
[nohup.txt](https://github.com/jagrosh/MusicBot/files/7183648/nohup.txt)
Funniest thing is there are no issues while all 3 servers are playing music, only when no one uses the bot.

**Reply by MichailiK** [reply]
```
[02:37:52] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
[02:44:36] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
[02:45:31] [WARN] [WebSocketClient]: Missed 2 heartbeats! Trying to reconnect...
[02:45:56] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
[02:47:17] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 4s
[02:48:00] [WARN] [WebSocketClient]: Got disconnected from WebSocket (Code 1000). Appending to reconnect queue
[02:48:09] [INFO] [WebSocketClient]: Connected to WebSocket
[02:48:10] [INFO] [JDA]: Finished (Re)Loading!
[04:24:56] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
[04:25:11] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 4s
```

After this, the bot doesn't attempt to reconnect at all? This seems weird. It's a very uneducated guess, but could Oracle perhaps "hibernating" your instance after some inactivity?

**Reply by kruleucc** [reply]
As far as I know, Oracle hibernates the console after certain time of inactivity, but it doesn't hibernate the instance, it is always running.
I'll try to add the bot to a server where no one does anything, and see if it goes offline in any point of the day.

As of now it only goes offline at night.

**Reply by kruleucc** [reply]
> ```
> [02:37:52] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
> [02:44:36] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
> [02:45:31] [WARN] [WebSocketClient]: Missed 2 heartbeats! Trying to reconnect...
> [02:45:56] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
> [02:47:17] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 4s
> [02:48:00] [WARN] [WebSocketClient]: Got disconnected from WebSocket (Code 1000). Appending to reconnect queue
> [02:48:09] [INFO] [WebSocketClient]: Connected to WebSocket
> [02:48:10] [INFO] [JDA]: Finished (Re)Loading!
> [04:24:56] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 2s
> [04:25:11] [WARN] [WebSocketClient]: Reconnect failed! Next attempt in 4s
> ```
> 
> After this, the bot doesn't attempt to reconnect at all? This seems weird. It's a very uneducated guess, but could Oracle perhaps "hibernating" your instance after some inactivity?

I'll run the bot on wsl on my pc tonight to see if Oracle is messing it up, and I'll post the details tommorow morning.

**Reply by kruleucc** [reply]
@MichailiK as you supposed, the Oracle is stopping the bot for no reason, just checked the logs after a whole night of hosting it on my pc, still running normally.

I might aswell get a Raspberry Pi to host it, that wouldn't be a bad idea.
